prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts

```diff
@@ -82,11 +105,62 @@ export class SiteManagementPageComponent implements OnInit {
   }
 
   public ngOnInit(): void {
+    this.createFormInstance();
+    this.initForm();
     const authorizedUser = this.authorizedUserService.authorizedUser;
     this.healthAuthorityId = authorizedUser.healthAuthorityCode;
-    this.healthAuthoritySites$ = this.authorizedUserResource.getAuthorizedUserSites(authorizedUser.id);
+    this.authorizedUserResource.getAuthorizedUserSites(authorizedUser.id)
+      .subscribe((sites: HealthAuthoritySiteList[]) => {
+        this.healthAuthoritySites = sites.sort((a, b) => a.siteName.toLocaleLowerCase().localeCompare(b.siteName.toLocaleLowerCase()))
+
+        const haVendors = this.healthAuthoritySites.map((s) => {
+          return s.healthAuthorityVendor.vendorCode;
+        });
+
+        const haCareTypes = this.healthAuthoritySites.map((s) => {
+          return s.healthAuthorityCareType.careType;
+        });
+
+        this.careTypes = [... new Set(haCareTypes)];
+        this.vendors = this.vendors.filter((v) => haVendors.some((hav) => hav === v.code));
+      });
   }
 
+  private createFormInstance() {
+    this.form = this.fb.group({
+      vendorCode: ['all', []],
+      careTypeCode: ['all', []],
+    });
+  }
+
+  private initForm() {
+    this.vendorCode.valueChanges.subscribe(() => {
+      this.filterSites();
+    });
+    this.careTypeCode.valueChanges.subscribe(() => {
+      this.filterSites();
+    })
+  }
+
+  public filterSites() {
+    const authorizedUser = this.authorizedUserService.authorizedUser;
+    this.authorizedUserResource.getAuthorizedUserSites(authorizedUser.id)
+      .subscribe((sites: HealthAuthoritySiteList[]) => {
+        this.healthAuthoritySites = sites.sort((a, b) => a.siteName.toLocaleLowerCase().localeCompare(b.siteName.toLocaleLowerCase()));
```

Comment:
A site in progress may not have a Site Name yet

Reply:
Fixed.